xwiki-platform-core/xwiki-platform-ckeditor/xwiki-platform-ckeditor-plugins/src/main/webjar/xwiki-table/plugin.js

```diff
@@ -69,15 +69,17 @@
     ['txtBorder', 'txtWidth', 'txtCellSpace', 'txtCellPad'].forEach(function(fieldId) {
       delete infoTab.get(fieldId)['default'];
     });
-    // Set a new default for the Headers value
+    // Set a new default for the Headers value.
     infoTab.get( 'selHeaders' )[ 'default' ] = 'row';
-    // Create a warning message for the accessibility without headers
-    // We navigate the structure of the macro to insert this message exactly next to the header.
+    // Create a warning message for the accessibility without headers.
+    const warningMessage = document.createElement('span');
+    warningMessage.classList.add('box', 'warningmessage');
+    warningMessage.textContent = CKEDITOR.tools.htmlEncode(
```

Comment:
There's no need to escape HTML when you set the ``textContent``. If you want to use ``CKEDITOR.tools.htmlEncode()`` then you can keep your previous code and just call ``CKEDITOR.tools.htmlEncode()`` on the translation value. So either use directly the DOM API (``textContent``) as I suggested, or use string concatenation but with HTML escaping.